server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -112,6 +113,16 @@ protected void doExecute(Task task, FieldCapabilitiesRequest request, final Acti
         checkIndexBlocks(clusterState, concreteIndices);
 
         final Map<String, FieldCapabilitiesIndexResponse> indexResponses = Collections.synchronizedMap(new HashMap<>());
+        final Map<String, Map<String, IndexFieldCapabilities>> indexMappingHashToResponses = Collections.synchronizedMap(new HashMap<>());
```

Comment:
It'd be great to add a comment explaining why we are using this shared map, it wasn't totally clear to me at first.

Reply:
I added a comment in
https://github.com/elastic/elasticsearch/pull/83494/commits/96abec0f5ab9da9823e7184767d0d07c0cc151ac.